ci/cron/src/Main.hs

```diff
@@ -245,13 +241,11 @@ fetch_gh_versions pred = do
 fetch_s3_versions :: DocOptions -> IO Versions
 fetch_s3_versions opts = do
     -- On the first run, this will fail so treat it like an empty file.
-    -- We could technically remove the catch later.
-    dropdown <- fetch "versions.json" False `catchIO`
-      (\_ -> pure [])
-    hidden <- fetch "hidden.json" True `catchIO`
-      (\_ -> pure [])
+    dropdown <- fetch "versions.json" False `or` []
+    hidden <- fetch "snapshots.json" True `or` []
     return $ versions $ dropdown <> hidden
-    where fetch file prerelease = do
+    where or io def = catchIO io (\_ -> pure def)
```

Comment:
not a huge fan of hiding names defined in `prelude` but I don’t have a great suggestion for another name. Maybe just keep it inlined?